src/community/features-autopopulate/features-autopopulate-core/src/main/java/org/geoserver/autopopulate/AutopopulateTransactionCallback.java

```diff
@@ -0,0 +1,439 @@
+                        }
+                        ((Update) element).setUpdateProperties(properties);
+                        transformed.getProperties().stream()
+                                .forEach(p -> log.info("Feature Property: " + p));
+                        ((Update) element)
+                                .getUpdateProperties().stream()
+                                        .forEach(
+                                                p ->
+                                                        log.info(
+                                                                "Update Property: "
+                                                                        + p.getName()
+                                                                        + " "
+                                                                        + p.getValue()));
+                    }
+                } catch (IOException e) {
+                    // Do never make the transaction fail due to an
+                    // AutopopulateTransactionCallback error.
+                    // Yell on the logs though
+                    log.log(
+                            Level.WARNING,
+                            "Error pre computing the transaction's affected attributes",
+                            e);
+                }
+                newElements.add(element);
+            } else if (element instanceof Delete) {
+                newElements.add(element);
+            }
+        }
+
+        // Replace the elements in the request
+        request.setElements(newElements);
+        return request;
+    }
+
+    /**
+     * Get the feature type info for the given feature type.
+     *
+     * @param featureType
+     * @return
+     */
+    private FeatureTypeInfo getFeatureTypeInfo(FeatureType featureType) {
+        Name featureTypeName = featureType.getName();
+        return getFeatureTypeInfo(featureTypeName);
+    }
+
+    /**
+     * Get the feature type info for the given feature type name.
+     *
+     * @param featureTypeName
+     * @return
+     */
+    private FeatureTypeInfo getFeatureTypeInfo(Name featureTypeName) {
+        FeatureTypeInfo featureTypeInfo = catalog.getFeatureTypeByName(featureTypeName);
+        if (featureTypeInfo == null) {
+            throw new RuntimeException(
+                    String.format("Couldn't find feature type info ''%s.", featureTypeName));
+        }
+        return featureTypeInfo;
+    }
+
+    /**
+     * Get the features that are affected by the given update.
+     *
+     * @param update
+     * @return
+     */
+    private SimpleFeatureCollection getTransactionFeatures(Update update) throws IOException {
+        QName typeName = update.getTypeName();
+        Filter filter = update.getFilter();
+        SimpleFeatureSource source = getTransactionSource(update);
+        try {
+            List<SimpleFeature> recent = DataUtilities.list(source.getFeatures(filter));
```

Comment:
Ouch no, the list of features affected by an Update might be... all of them. This is inviting a OOM to happen.
Also see my comment above about avoiding scanning the updated features at all.

Reply:
Removed.